pyclam/manifold.py

```diff
@@ -308,18 +308,29 @@ def partition(self, *criterion) -> Iterable['Cluster']:
             self.manifold.data[farthest],
         ])
 
-        p1_idx, p2_idx = list(), list()
+        p1_idx, p2_idx = [self.argradius], [farthest]
         [(p1_idx if p1 < p2 else p2_idx).append(i)
          for batch in iter(self)
          for i, p1, p2 in zip(batch, *cdist(poles, self.manifold.data[batch], self.metric))]
 
+        # TODO: Neither of these should be possible
```

Comment:
Then why are we testing for them? bugs elsewhere?

Reply:
I don't know why this invariant breaks. I'll look into it after unifying distance calls.